packages/apollo-server-hapi/src/ApolloServer.ts

```diff
@@ -90,66 +93,99 @@ export class ApolloServer extends ApolloServerBase {
+            query:
+              request.method === 'post'
+                ? // TODO type payload as string or Record
+                  (request.payload as any)
+                : request.query,
+            request: convertNodeHttpToRequest(request.raw.req),
+          },
+        );
+
+        const response = h.response(graphqlResponse);
+        if (responseInit.headers) {
+          Object.entries(responseInit.headers).forEach(([headerName, value]) =>
+            response.header(headerName, value),
           );
+        }
+        response.code(responseInit.status || 200);
+        return response;
+      } catch (error) {
+        if (!isHttpQueryError(error)) {
+          throw Boom.boomify(error as Error);
+        }
 
-          const response = h.response(graphqlResponse);
-          if (responseInit.headers) {
-            Object.entries(responseInit.headers).forEach(
-              ([headerName, value]) => response.header(headerName, value),
-            );
-          }
-          response.code(responseInit.status || 200);
-          return response;
-        } catch (error) {
-          if (!isHttpQueryError(error)) {
-            throw Boom.boomify(error as Error);
-          }
-
-          if (true === error.isGraphQLError) {
-            const response = h.response(error.message);
-            if (error.headers) {
-              Object.entries(error.headers).forEach(([headerName, value]) => {
-                response.header(headerName, value);
-              });
-            }
-            response.code(error.statusCode);
-            response.type('application/json');
-            return response;
-          }
-
-          const err = new Boom.Boom(error.message, {
-            statusCode: error.statusCode,
-          });
+        if (true === error.isGraphQLError) {
+          const response = h.response(error.message);
           if (error.headers) {
             Object.entries(error.headers).forEach(([headerName, value]) => {
-              err.output.headers[headerName] = value;
+              response.header(headerName, value);
             });
           }
-          // Boom hides the error when status code is 500
-          err.output.payload.message = error.message;
-          throw err;
+          response.code(error.statusCode);
+          response.type('application/json');
+          return response;
         }
-      },
+
+        const err = new Boom.Boom(error.message, {
+          statusCode: error.statusCode,
+        });
+        if (error.headers) {
+          Object.entries(error.headers).forEach(([headerName, value]) => {
+            err.output.headers[headerName] = value;
+          });
+        }
+        // Boom hides the error when status code is 500
+        err.output.payload.message = error.message;
+        throw err;
+      }
+    };
+
+    // POST ROUTE
+    let postOptions = Object.assign({}, routePost);
+    postOptions = Object.assign(postOptions, route);
```

Comment:
Doesn't this mean that things on `route` will take precedence over things on routePost if there are conflicts? Seems a bit backwards.

Also it's a bit easier to write `{...routePost, ...route}` (or the reverse if I'm correct) rather than use Object.assign these days.

Reply:
You are correct, I swapped to routePost / routeGet overriding route options.